APE26.rst

```diff
@@ -0,0 +1,220 @@
+objects such as ``ICRS``, when the docs are clear that these are for more advanced users 
+and that ``SkyCoord`` is to be preferred. The system will also be less fragile. For 
+example, if users manipulate the internal workings of ``SkyCoord`` (which is discouraged 
+but possible), the coordinate data can become decoupled from the caching that ``SkyCoord`` 
+performs for speed.
+
+Finished Product
+----------------
+The end result of the implementation of this APE will be that coordinate frame classes 
+only hold information pertaining to the reference frame they represent and never actual 
+coordinate data in that reference frame. This is consistent with our mathematical 
+framework, the reference frame mediates how coordinate data is understood (e.g. distance 
+measures) or interacts (e.g. separation from other coordinates), but the coordinate data 
+itself is actually independent of that information. Classes like ``SkyCoord`` will be 
+composed structures bringing together the reference frame (an instance of a 
+``BaseCoordinateFrame`` subclass) and the coordinate data (``Representation`` objects).
+
+We illustrate this with the following pseudocode.
+
+.. code-block:: python
+
+    @dataclass
+    class BaseCoordinateFrame:
+        ...
+
+    @dataclass
+    class FK5(BaseCoordinateFrame):
+        equinox: Time
+
+    class SkyCoord:
+        frame: BaseCoordinateFrame
+        data: Representation
+
+        def __init__(...):
+            ...
+
+Branches and pull requests
+--------------------------
+No direct progress on these changes has yet occurred. Discussion of these ideas has 
+however arisen in multiple issues and pull requests, demonstrating the need for and 
+utility of the proposed changes.
+
+Several issues have been raised regarding topics such as confusion differentiating the 
+use of ``frame`` and ``SkyCoord`` for data storage, and problems arising in other astropy 
+subpackages when using frames that store data. For example:
+
+- *Comparing Frame with data and SkyCoord with same data raises exception*: `https://github.com/astropy/astropy/issues/13476`
+- *Add Frame objects without data to a Table*: `https://github.com/astropy/astropy/issues/16823`
+
+Additionally, multiple pull requests have factored out common code between frames and 
+``SkyCoord``, showing that there is no proper separation of concern:
+
+- *Introduction of shared `CoordinateFrameInfo`*: `https://github.com/astropy/astropy/pull/16831`
+- *Introduction of `CoordinateSharedMethods`* (this was later removed and instead methods 
+were duplicated): `https://github.com/astropy/astropy/pull/17016`
+
+Further, pull requests have added methods to make frames and ``SkyCoord`` even more 
+similar, underscoring that frames *with* data should not be separate entities from 
+``SkyCoord``:
+
+- *Add .to_table() to frames*: `https://github.com/astropy/astropy/pull/17009`
+- *Add .frame attribute to frames*: `https://github.com/astropy/astropy/pull/16356`
```

Comment:
I find it confusing that you have not used the pull request titles.

Reply:
Good point, thank you. Fixed in [46f503e](https://github.com/astropy/astropy-APEs/pull/112/commits/46f503ee97e7ec3f1d050b29b6949a52a48272c2)